ui/helpers/utils/export-utils.js

```diff
@@ -1,11 +1,93 @@
-import { getRandomFileName } from './util';
+/**
+ * @enum { string }
+ */
+export const ExportableContentType = {
+  JSON: 'application/json',
+  TXT: 'text/plain',
+};
 
-export function exportAsFile(filename, data, type = 'text/csv') {
+/**
+ * @enum { string }
+ */
+const ExtensionForContentType = {
+  [ExportableContentType.JSON]: '.json',
+  [ExportableContentType.TXT]: '.txt',
+};
+
+/**
+ * Export data as a file.
+ *
+ * @param {string} filename - The name of the file to export.
+ * @param {string} data - The data to export.
+ * @param {ExportableContentType} contentType - The content type of the file to export.
+ */
+export async function exportAsFile(filename, data, contentType) {
+  if (!ExtensionForContentType[contentType]) {
+    throw new Error(`Unsupported file type: ${contentType}`);
+  }
+
+  if (supportsShowSaveFilePicker()) {
+    // Preferred method for downloads
+    await saveFileUsingFilePicker(filename, data, contentType);
+  } else {
+    saveFileUsingDataUri(filename, data, contentType);
+  }
+}
+/**
+ * Notes if the browser supports the File System Access API.
+ *
+ * @returns {boolean}
+ */
+function supportsShowSaveFilePicker() {
+  return (
+    typeof window !== 'undefined' &&
+    typeof window.showSaveFilePicker !== 'undefined'
+  );
+}
+
+/**
+ * Saves a file using the File System Access API.
+ *
+ * @param {string} filename - The name of the file to export.
+ * @param {string} data - The data to export.
+ * @param {ExportableContentType} contentType - The content type of the file to export.
+ * @returns {Promise<void>}
+ */
+async function saveFileUsingFilePicker(filename, data, contentType) {
+  // eslint-disable-next-line no-undef
```

Comment:
Nit: rather than disable this lint warning, we can add it to our ESLint config list of globals (I think this is the right place: https://github.com/MetaMask/metamask-extension/blob/4b2c777862c0455a0f62a6239ca7056b609fd46d/.eslintrc.base.js#L9 )

Reply:
Or maybe we could reference it as `window.Blob`